### DISCUSSION TITLE
Geojson in Layers menu

### DISCUSSION THREAD

**Discussion by oyvindsolstad** [javalent/obsidian-leaflet#263] [Ideas]
Hi, after the tileServer og tileOverlays [feature](https://github.com/valentine195/obsidian-leaflet-plugin/compare/4.5.0...4.6.0) came, I noticed that you also can turn on and off geojson-files in the menu, that are linked in the code. BUT - these files do not keep their names. They show up in the menu as geojson1, 2 etc. 

Really a small thing, but would it be possible to either have them keep the name they have in the code, or alt option: be able to name them with a | as with tileServers and tileOverlays? 

![Skjermbilde 2021-12-16 kl  17 42 39](https://user-images.githubusercontent.com/4631971/146412697-0150d391-e763-4c59-96f8-50a1b46b0cd6.png)

Code:

```
geojson:
  - [[I-80.json]]
  - [[I-90.json]]
  - [[I-40.json]]
  - [[I-70.json]]
  - [[I-2.json]]
  - [[I-4.json]]
  - [[I-10.json]]
  - [[I-27.json]]
  - [[I-95.json]]
  - [[geo-usa-route66-edit.json]]
```

**Comment by valentine195** [comment]
Hey, sorry, I missed this.

> be able to name them with a | as with tileServers and tileOverlays?

You can do this.

**Reply by claremacrae** [reply]
That's fabulous! It would be great to see that in the documentation somewhere...

Is there any particular reason why it doesn't use the filename as the label by default, if no `|blah` is provided, like in the following, it would be nice not to have to duplicate the text?

```
geojson: [[Route-USA-2021.json|Route-USA-2021]]
```

**Reply by valentine195** [reply]
The only reason was because my test files had some ugly names :) if you would prefer the other behavior I can do that instead.

I did add this to the docs when I saw this, must’ve slipped through.

**Reply by valentine195** [reply]
> Is there any particular reason why it doesn't use the filename as the label by default

This is now the default behavior in 4.12.1

**Reply by claremacrae** [reply]
> > Is there any particular reason why it doesn't use the filename as the label by default
> 
> This is now the default behavior in 4.12.1

Wow that's amazing - thank you!

I see it includes the `.json` suffix in the menu label - I think that's harmless.

**Reply by oyvindsolstad** [reply]
Great! Thank you!
